diff -Naurp linux-amlogic-3.10-c954850/arch/arm/boot/dts/amlogic/meson8m2_n200C_2G.dtd linux-amlogic-3.10-c954850-patch/arch/arm/boot/dts/amlogic/meson8m2_n200C_2G.dtd
--- linux-amlogic-3.10-c954850/arch/arm/boot/dts/amlogic/meson8m2_n200C_2G.dtd	2016-02-27 02:12:06.000000000 +0200
+++ linux-amlogic-3.10-c954850-patch/arch/arm/boot/dts/amlogic/meson8m2_n200C_2G.dtd	2016-05-12 15:52:50.000000000 +0300
@@ -88,7 +88,7 @@ void root_func(){
 
         cma_0:region@0 {
            region_name = "cma_0";
-           reg = <0 0x01200000>;
+           reg = <0 0x02400000>;
            linux,contiguous-region;
         };
 	};
@@ -133,11 +133,11 @@ void root_func(){
 		compatible = "amlogic,mesonfb";
 		dev_name = "mesonfb";
 		status = "okay";
-		reserve-memory = <0x01800000  0x00100000>;
+		reserve-memory = <0x04800000  0x00100000>;
 		reserve-iomap = "true";
                 vmode = <3>; /**0:VMODE_720P 1:VMODE_LCD  2:VMODE_LVDS_1080P 3:VMODE_1080P*/
 		scale_mode = <1>; /*0:default 1:new*/
-		4k2k_fb = <0>;
+		4k2k_fb = <1>;
 		display_size_default = <1920 1080 1920 3240 32>; //1920*1080*4*3 = 0x17BB000
 	};
 
@@ -148,7 +148,7 @@ void root_func(){
 		compatible = "amlogic,deinterlace";
 		dev_name = "deinterlace";
 		status = "ok";
-		reserve-memory = <0x02100000>; //10x1920x1088x3/2=33M
+		reserve-memory = <0x01800000>; //8x1920x1088x3/2=24M
 	};
 
 /// ***************************************************************************************
@@ -175,22 +175,9 @@ void root_func(){
                 compatible = "amlogic,vdec";
                 dev_name = "vdec.0";
                 status = "okay";
-								reserve-memory = <0x04000000>; // 64M
-//		reserve-iomap = "true";
+                reserve-memory = <0x10000000>;
+                reserve-iomap = "true";
         };
-///	-	picture decoder
-//$$ MODULE="picture decoder"
-//$$ DEVICE="picdec"
-//$$ L2 PROP_STR = "status"
-//$$ L3 PROP_U32 2 ="reg"
-    picdec{
-		compatible = "amlogic,picdec";
-		dev_name = "picdec";
-		status = "okay";
-                share-memory-name = "vdec0";
-                share-memory-offset = <0>;
-                share-memory-size = <0x04000000>; //64M
-	};
 
 /// ***************************************************************************************
 ///	-	DISP&MM-PostProcess
@@ -669,19 +656,19 @@ sdio{
 //$$ L3 PROP_U32 2 ="reg"
 //$$ L3 PROP_STR = "pinctrl-names"
 //$$ L2 PROP_CHOICE "I2C_AO_pin_match" = "pinctrl-0"
-	i2c@c8100500{ /*I2C-AO*/
-		compatible = "amlogic,aml_i2c";
-		dev_name = "i2c-AO";
-		status = "ok";
-		reg = <0xc8100500 0x1d>;
-		#address-cells = <1>;
-		#size-cells = <0>;
-		device_id = <0>;
-		pinctrl-names="default";
-		pinctrl-0=<&ao_i2c_master>;
-		use_pio = <0>;
-		master_i2c_speed = <100000>;
-	};
+//	i2c@c8100500{ /*I2C-AO*/
+//		compatible = "amlogic,aml_i2c";
+//		dev_name = "i2c-AO";
+//		status = "ok";
+//		reg = <0xc8100500 0x1d>;
+//		#address-cells = <1>;
+//		#size-cells = <0>;
+//		device_id = <0>;
+//		pinctrl-names="default";
+//		pinctrl-0=<&ao_i2c_master>;
+//		use_pio = <0>;
+//		master_i2c_speed = <100000>;
+//	};
 
 //$$ DEVICE = "I2C_A"
 //$$ L2 PROP_STR = "status"
@@ -1209,18 +1196,18 @@ sdio{
 //$$ L2 PROP_STR = "key_pin"
 //$$ L2 PROP_U32 = "irq_keyup"
 //$$ L2 PROP_U32 = "irq_keydown"
-//    gpio_keypad{
-//		compatible = "amlogic,gpio_keypad";
-//		status = "okay";
-//		scan_period = <20>;
-//		key_num = <1>;
-//        key_name = "power";
-//		key_code = <116>;
-//		key_pin = "GPIOAO_3";
-//                gpio_high_z;
-//        irq_keyup = <6>;
-//        irq_keydown = <7>;
-//	};
+      gpio_keypad{
+	    compatible = "amlogic,gpio_keypad";
+	    status = "okay";
+	    scan_period = <20>;
+	    key_num = <1>;
+	    key_name = "power";
+	    key_code = <116>;
+	    key_pin = "GPIOAO_4";
+	    gpio_high_z;
+	    irq_keyup = <6>;
+	    irq_keydown = <7>;
+	};
 //$$ DEVICE="saradc"
 //$$ L2 PROP_STR = "status"
 	saradc{
@@ -1856,42 +1843,6 @@ sdio{
 			amlogic,pins = "BOOT_11","BOOT_12","BOOT_13","BOOT_18";
 		};
 
-//$$ MATCH "dvb_p_ts0_pins_match" = "&dvb_p_ts0_pins"
-//$$ L2 PROP_U32 2 = "amlogic,setmask"
-//$$ L2 PROP_U32 8 = "amlogic,clrmask"
-//$$ L2 PROP_STR 12 = "amlogic,pins"
-		dvb_p_ts0_pins: dvb_p_ts0_pins {
-			amlogic,setmask = <3 0x3F>;
-			amlogic,clrmask = <1 0xCCE41
-								4 0x2000000
-								7 0x20000000
-								9 0x431F>;
-			amlogic,pins = "GPIOY_9","GPIOY_10","GPIOY_11","GPIOY_12","GPIOY_13","GPIOY_14","GPIOY_15","GPIOY_16","GPIOY_0","GPIOY_1","GPIOY_4","GPIOY_8";
-		};
-
-//$$ MATCH "dvb_s_ts0_pins_match" = "&dvb_s_ts0_pins"
-//$$ L2 PROP_U32 2 = "amlogic,setmask"
-//$$ L2 PROP_U32 6 = "amlogic,clrmask"
-//$$ L2 PROP_STR 5 = "amlogic,pins"
-		dvb_s_ts0_pins: dvb_s_ts0_pins {
-			amlogic,setmask = <3 0x1F>;
-			amlogic,clrmask = <1 0xCCE41
-								4 0x2000000
-								9 0x310>;
-			amlogic,pins = "GPIOY_9","GPIOY_0","GPIOY_1","GPIOY_4","GPIOY_8";
-		};
-
-//$$ MATCH "dvb_fe_pins_match" = "&dvb_fe_pins"
-//$$ L2 PROP_U32 8 = "amlogic,clrmask"
-//$$ L2 PROP_STR 1 = "amlogic,pins"
-		dvb_fe_pins: dvb_fe_pins {
-			amlogic,clrmask = <1 0x38
-								3 0x18000
-								4 0x1800000
-								9 0xC0>;
-			amlogic,pins = "GPIOY_5","GPIOY_6";
-		};
-
 //$$ MATCH "meson_vcck_dvfs_pin_0_match" = "&aml_pwm_pins"
 //$$ L2 PROP_U32 2 = "amlogic,setmask"
 //$$ L2 PROP_U32 2 = "amlogic,clrmask"
@@ -1918,53 +1869,6 @@ sdio{
         };
 
 /// ***************************************************************************************
-///	-	DVB
-//$$ MODULE="DVB"
-
-//$$ DEVICE="dvb"
-//$$ L2 PROP_STR = "ts0"
-//$$ L2 PROP_U32 = "ts0_invert"
-//$$ L2 PROP_U32 = "ts0_control"
-//$$ L2 PROP_STR 6 = "pinctrl-names"
-//$$ L2 PROP_CHOICE "dvb_p_ts0_pins_match" = "pinctrl-0"
-//$$ L2 PROP_CHOICE "dvb_s_ts0_pins_match" = "pinctrl-1"
-	dvb{
-		compatible = "amlogic,dvb";
-		/*"parallel","serial","disable"*/
-		ts0 = "parallel";
-		ts0_control = <0>;
-		ts0_invert = <0>;
-		pinctrl-names = "p_ts0", "s_ts0";
-		pinctrl-0 = <&dvb_p_ts0_pins>;
-		pinctrl-1 = <&dvb_s_ts0_pins>;
-	};
-
-//$$ DEVICE="dvbfe"
-//$$ L2 PROP_STR = "dtv_demod0"
-//$$ L2 PROP_U32 = "dtv_demod0_i2c_adap_id"
-//$$ L2 PROP_U32 = "dtv_demod0_i2c_addr"
-//$$ L2 PROP_U32 = "dtv_demod0_reset_value"
-//$$ L2 PROP_STR = "dtv_demod0_reset_gpio"
-//$$ L2 PROP_U32 = "fe0_dtv_demod"
-//$$ L2 PROP_U32 = "fe0_ts"
-//$$ L2 PROP_U32 = "fe0_dev"
-//$$ L2 PROP_STR 6 = "pinctrl-names"
-//$$ L2 PROP_CHOICE "dvb_fe_pins_match" = "pinctrl-0"
-	dvbfe{
-		compatible = "amlogic,dvbfe";
-		dtv_demod0 = "Avl6211";
-		dtv_demod0_i2c_adap_id = <2>;
-		dtv_demod0_i2c_addr = <0xC0>;
-		dtv_demod0_reset_value = <0>;
-		dtv_demod0_reset_gpio = "GPIOY_5";
-		fe0_dtv_demod = <0>;
-		fe0_ts = <0>;
-		fe0_dev = <0>;
-		pinctrl-names = "default";
-		pinctrl-0 = <&dvb_fe_pins>;
-	};
-
-/// ***************************************************************************************
 ///	-	RNG
 //$$ MODULE="RNG"
 //$$ DEVICE="meson-rng"
