diff -Naurp tvh-e4e96ff/src/input/mpegts/linuxdvb/linuxdvb_frontend.c tvh-e4e96ff-patch/src/input/mpegts/linuxdvb/linuxdvb_frontend.c
--- tvh-e4e96ff/src/input/mpegts/linuxdvb/linuxdvb_frontend.c	2019-02-10 11:26:33.000000000 +0200
+++ tvh-e4e96ff-patch/src/input/mpegts/linuxdvb/linuxdvb_frontend.c	2019-02-10 11:30:50.000000000 +0200
@@ -138,7 +138,7 @@ const idclass_t linuxdvb_frontend_class
       .desc     = N_("The limit for the PID filter (driver or hardware)."),
       .off      = offsetof(linuxdvb_frontend_t, lfe_pids_max),
       .opts     = PO_ADVANCED,
-      .def.i    = 32
+      .def.i    = 26
     },
     {
       .type     = PT_BOOL,
@@ -2165,7 +2165,7 @@ linuxdvb_frontend_create
   lfe->lfe_name   = strdup(name);
   lfe->lfe_ibuf_size = 188000;
   lfe->lfe_status_period = 1000;
-  lfe->lfe_pids_max = 32;
+  lfe->lfe_pids_max = 26;
   lfe->lfe_pids_use_all = 1;
   lfe->lfe_sig_multiplier = 100;
   lfe->lfe_snr_multiplier = 100;
