diff -Naurp tvheadend-77fc1c0/src/webui/webui.c tvheadend-77fc1c0-patch/src/webui/webui.c
--- tvheadend-77fc1c0/src/webui/webui.c	2018-10-30 11:09:17.000000000 +0200
+++ tvheadend-77fc1c0-patch/src/webui/webui.c	2018-10-30 11:14:37.000000000 +0200
@@ -519,7 +519,7 @@ http_m3u_playlist_add(htsbuf_queue_t *hq
   if (!strempty(logo)) {
     int id = imagecache_get_id(logo);
     if (id) {
-      htsbuf_qprintf(hq, " logo=\"%s/imagecache/%d", hostpath, id);
+      htsbuf_qprintf(hq, " tvg-logo=\"%s/imagecache/%d", hostpath, id);
       switch (urlauth) {
       case URLAUTH_NONE:
         break;
@@ -533,7 +533,7 @@ http_m3u_playlist_add(htsbuf_queue_t *hq
       }
       htsbuf_append_str(hq, "\"");
     } else {
-      htsbuf_qprintf(hq, " logo=\"%s\"", logo);
+      htsbuf_qprintf(hq, " tvg-logo=\"%s\"", logo);
     }
   }
   if (epgid)
