diff -Naurp linux-1be06c2/arch/arm/mach-meson8b/pm.c linux-1be06c2-patch/arch/arm/mach-meson8b/pm.c
--- linux-1be06c2/arch/arm/mach-meson8b/pm.c	2019-02-06 11:28:18.000000000 +0200
+++ linux-1be06c2-patch/arch/arm/mach-meson8b/pm.c	2019-02-06 11:41:54.000000000 +0200
@@ -20,6 +20,7 @@
 #include <asm/cacheflush.h>
 #include <asm/delay.h>
 #include <asm/uaccess.h>
+#include <linux/gpio.h>
 
 #include <mach/pm.h>
 #include <mach/am_regs.h>
@@ -36,6 +37,7 @@
 #include <mach/meson-secure.h>
 #endif
 
+#include <linux/amlogic/aml_gpio_consumer.h>
 #include <mach/register.h>
 #include <mach/watchdog.h>
 #include <mach/mod_gate.h>
@@ -431,6 +433,7 @@ static void meson_pm_suspend(void)
 #ifdef CONFIG_MESON_TRUSTZONE
 		meson_suspend_firmware();
 #else
+		amlogic_set_value(GPIOAO_1,0,"power_led");
 		meson_power_suspend();
 #endif
 #else
@@ -547,6 +550,7 @@ static struct meson_pm_config aml_pm_pda
 
 static int __init meson_pm_probe(struct platform_device *pdev)
 {
+	int ret;
 	printk(KERN_INFO "enter meson_pm_probe!\n");
 #ifdef CONFIG_HAS_EARLYSUSPEND
 	early_suspend.level = EARLY_SUSPEND_LEVEL_DISABLE_FB;
@@ -554,6 +558,13 @@ static int __init meson_pm_probe(struct
 	early_suspend.resume = meson_system_late_resume;
 	register_early_suspend(&early_suspend);
 #endif
+    ret = amlogic_gpio_request_one(GPIOAO_1,GPIOF_OUT_INIT_HIGH,"power_led"); //for power LED
+        if(ret){
+            printk("---%s----can not set output pin \n",__func__);
+            amlogic_gpio_free(GPIOAO_1,"power_led");
+        }else{
+            printk("---%s----set output pin success\n",__func__);
+        }
 	pdev->dev.platform_data=&aml_pm_pdata;
 	pdata = pdev->dev.platform_data;
 	if (!pdata) {
