diff -Naurp kodi-bdfd398/addons/resource.language.en_gb/resources/strings.po kodi-bdfd398-patch/addons/resource.language.en_gb/resources/strings.po
--- kodi-bdfd398/addons/resource.language.en_gb/resources/strings.po	2019-02-02 11:05:11.000000000 +0200
+++ kodi-bdfd398-patch/addons/resource.language.en_gb/resources/strings.po	2019-02-02 12:01:15.000000000 +0200
@@ -6012,7 +6012,7 @@ msgstr ""
 
 #: system/settings/settings.xml
 msgctxt "#13009"
-msgid "Quit"
+msgid "TVIP"
 msgstr ""
 
 #: system/settings/settings.xml
@@ -6031,7 +6031,7 @@ msgstr ""
 #: addons/skin.estuary/xml/DialogButtonMenu.xml
 #: addons/skin.estouchy/xml/Home.xml
 msgctxt "#13012"
-msgid "Exit"
+msgid "TVIP"
 msgstr ""
 
 #: addons/skin.estuary/xml/DialogButtonMenu.xml
diff -Naurp kodi-bdfd398/xbmc/Application.cpp kodi-bdfd398-patch/xbmc/Application.cpp
--- kodi-bdfd398/xbmc/Application.cpp	2019-02-02 11:05:11.000000000 +0200
+++ kodi-bdfd398-patch/xbmc/Application.cpp	2019-02-02 11:32:37.000000000 +0200
@@ -198,6 +198,9 @@
 #include "addons/AddonSystemSettings.h"
 #include "FileItem.h"
 
+#include <fstream>
+#include <fcntl.h>
+
 using namespace ADDON;
 using namespace XFILE;
 #ifdef HAS_DVD_DRIVE
@@ -2021,7 +2024,8 @@ void CApplication::OnApplicationMessage(
     break;
 
   case TMSG_QUIT:
-    SetExitCode(EXITCODE_QUIT);
+    SetExitCode(EXITCODE_TVIP);
+    Stop(EXITCODE_TVIP);
     break;
 
   case TMSG_SHUTDOWN:
@@ -2643,6 +2647,14 @@ void CApplication::Stop(int exitCode)
 
   cleanup_emu_environ();
 
+  // start TVIP
+  if (m_ExitCode == EXITCODE_TVIP)
+  {
+    remove("/var/lock/start.kodi");
+    creat("/var/lock/start.tvip", S_IRUSR|S_IROTH);
+    system("systemctl start tvip.service");
+  }
+
   Sleep(200);
 }
 
diff -Naurp kodi-bdfd398/xbmc/guilib/guiinfo/SystemGUIInfo.cpp kodi-bdfd398-patch/xbmc/guilib/guiinfo/SystemGUIInfo.cpp
--- kodi-bdfd398/xbmc/guilib/guiinfo/SystemGUIInfo.cpp	2019-02-02 11:05:11.000000000 +0200
+++ kodi-bdfd398-patch/xbmc/guilib/guiinfo/SystemGUIInfo.cpp	2019-02-02 11:41:21.000000000 +0200
@@ -565,7 +565,7 @@ bool CSystemGUIInfo::GetBool(bool& value
       value = !(CServiceBroker::GetGUI()->GetWindowManager().GetActiveWindow() == WINDOW_LOGIN_SCREEN);
       return true;
     case SYSTEM_SHOW_EXIT_BUTTON:
-      value = CServiceBroker::GetSettingsComponent()->GetAdvancedSettings()->m_showExitButton;
+      value = true;
       return true;
     case SYSTEM_HAS_LOGINSCREEN:
       value = CServiceBroker::GetSettingsComponent()->GetProfileManager()->UsingLoginScreen();
diff -Naurp kodi-bdfd398/xbmc/powermanagement/PowerManager.cpp kodi-bdfd398-patch/xbmc/powermanagement/PowerManager.cpp
--- kodi-bdfd398/xbmc/powermanagement/PowerManager.cpp	2019-01-31 13:58:11.000000000 +0200
+++ kodi-bdfd398-patch/xbmc/powermanagement/PowerManager.cpp	2019-02-02 12:02:10.000000000 +0200
@@ -264,17 +264,6 @@ void CPowerManager::RestorePlayerState()
 
 void CPowerManager::SettingOptionsShutdownStatesFiller(SettingConstPtr setting, std::vector< std::pair<std::string, int> > &list, int &current, void *data)
 {
-  if (CServiceBroker::GetPowerManager().CanPowerdown())
-    list.push_back(make_pair(g_localizeStrings.Get(13005), POWERSTATE_SHUTDOWN));
-  if (CServiceBroker::GetPowerManager().CanHibernate())
-    list.push_back(make_pair(g_localizeStrings.Get(13010), POWERSTATE_HIBERNATE));
-  if (CServiceBroker::GetPowerManager().CanSuspend())
-    list.push_back(make_pair(g_localizeStrings.Get(13011), POWERSTATE_SUSPEND));
-  if (!g_application.IsStandAlone())
-  {
-    list.push_back(make_pair(g_localizeStrings.Get(13009), POWERSTATE_QUIT));
-#if !defined(TARGET_DARWIN_IOS)
-    list.push_back(make_pair(g_localizeStrings.Get(13014), POWERSTATE_MINIMIZE));
-#endif
-  }
+  list.push_back(make_pair(g_localizeStrings.Get(13005), POWERSTATE_SHUTDOWN));
+  list.push_back(make_pair(g_localizeStrings.Get(13009), POWERSTATE_QUIT));
 }
diff -Naurp kodi-bdfd398/xbmc/XBApplicationEx.h kodi-bdfd398-patch/xbmc/XBApplicationEx.h
--- kodi-bdfd398/xbmc/XBApplicationEx.h	2019-02-02 11:05:11.000000000 +0200
+++ kodi-bdfd398-patch/xbmc/XBApplicationEx.h	2019-02-02 11:34:29.000000000 +0200
@@ -18,6 +18,7 @@ enum {
   EXITCODE_POWERDOWN = 64,
   EXITCODE_RESTARTAPP= 65,
   EXITCODE_REBOOT    = 66,
+  EXITCODE_TVIP      = 67,
 };
 
 class CXBApplicationEx : public IWindowManagerCallback
