From 53723f8c0a13848a1c7241cf4b5e9eb6aeb60898 Mon Sep 17 00:00:00 2001
From: SqTER-PL <26832944+SqTER-PL@users.noreply.github.com>
Date: Thu, 25 Jul 2019 09:49:13 +0200
Subject: [PATCH] Fix detect AC3+ audio track (TVH_4.2), fixes#5684

---
 src/input/mpegts/dvb_psi.c | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/src/input/mpegts/dvb_psi.c b/src/input/mpegts/dvb_psi.c
index cafa3def3f..8ef13383b2 100644
--- a/src/input/mpegts/dvb_psi.c
+++ b/src/input/mpegts/dvb_psi.c
@@ -2425,6 +2425,10 @@ psi_parse_pmt
     case 0x81:
       hts_stream_type = SCT_AC3;
       break;
+          
+    case 0x87:
+      hts_stream_type = SCT_EAC3;
+      break;
     
     case 0x0f:
       hts_stream_type = SCT_MP4A;
@@ -2507,7 +2511,7 @@ psi_parse_pmt
         break;
 
       case DVB_DESC_EAC3:
-        if(estype == 0x06 || estype == 0x81)
+        if(estype == 0x06 || estype == 0x87)
           hts_stream_type = SCT_EAC3;
         break;
 
